tiledb/sm/serialization/query.cc

```diff
@@ -805,14 +958,42 @@ Status query_to_capnp(Query& query, capnp::Query::Builder* query_builder) {
   query_builder->setTotalValidityBufferBytes(total_validity_len_bytes);
 
   if (type == QueryType::READ) {
-    auto builder = query_builder->initReader();
-    auto reader = (Reader*)query.strategy();
-
-    query_builder->setVarOffsetsMode(reader->offsets_mode());
-    query_builder->setVarOffsetsAddExtraElement(
-        reader->offsets_extra_element());
-    query_builder->setVarOffsetsBitsize(reader->offsets_bitsize());
-    RETURN_NOT_OK(reader_to_capnp(query, *reader, &builder));
+    bool found = false;
+    bool use_refactored_readers = false;
+    RETURN_NOT_OK(query.config()->get<bool>(
```

Comment:
Is there a more permanent way to detect this call path? I'm considered about the future when we some of the refactored readers become permanent but we might still have "legacy" readers which need the `else` sections.